rfc/rfc-51/rfc-51.md

```diff
@@ -64,69 +65,72 @@ We follow the debezium output format: four columns as shown below
 
 Note: the illustration here ignores all the Hudi metadata columns like `_hoodie_commit_time` in `before` and `after` columns.
 
-## Goals
+## Design Goals
 
 1. Support row-level CDC records generation and persistence;
 2. Support both MOR and COW tables;
 3. Support all the write operations;
 4. Support Spark DataFrame/SQL/Streaming Query;
 
-## Implementation
+## Configurations
 
-### CDC Architecture
+| key                                                 | default  | description                                                                                                                                      |
+|-----------------------------------------------------|----------|--------------------------------------------------------------------------------------------------------------------------------------------------|
+| hoodie.table.cdc.enabled                            | `false`  | The master switch of the CDC features. If `true`, writers and readers will respect CDC configurations and behave accordingly.                    |
+| hoodie.table.cdc.supplemental.logging               | `false`  | If `true`, persist the required information about the changed data, including `before`. If `false`, only `op` and record keys will be persisted. |
+| hoodie.table.cdc.supplemental.logging.include_after | `false`  | If `true`, persist `after` as well.                                                                                                              |
 
-![](arch.jpg)
+To perform CDC queries, users need to set `hoodie.table.cdc.enable=true` and `hoodie.datasource.query.type=incremental`.
 
-Note: Table operations like `Compact`, `Clean`, `Index` do not write/change any data. So we don't need to consider them in CDC scenario.
- 
-### Modifiying code paths
+| key                                    | default    | description                          |
+|----------------------------------------|------------|--------------------------------------|
+| hoodie.table.cdc.enabled               | `false`    | set to `true` for CDC queries        |
+| hoodie.datasource.query.type           | `snapshot` | set to `incremental` for CDC queries |
+| hoodie.datasource.read.start.timestamp | -          | requried.                            |
+| hoodie.datasource.read.end.timestamp   | -          | optional.                            |
 
-![](points.jpg)
+### Logical File Types
 
-### Config Definitions
+We define 4 logical file types for the CDC scenario.
```

Comment:
@xushiyan actually 5 file types. do you want to deal with `MOR_LOG_FILE` after we make an agreement about the cdc query on mor tables between flink and spark?

Reply:
sure. on the other hand, i'm actually not so sure how helpful to introduce these definitions of logical file types. i felt they are understanding overhead. What do you think? i'm more inclined to removing them.

Reply:
I think keeping them can help the developers to understand this design details.

Reply:
What is the logical file type here ? To help understand the work flow of writer/reader ?

Reply:
Can someone give a explanation here ? What is exactly a logical file type ? name it action seems more suitable here?

Reply:
These Logical file types are meant to group files/filesgroups logically based on different write scenarios specific for producing CDC data. I had similar concern on understanding these concepts, which are not generic hudi concepts. we should just list out a table here to map Hudi actions -> what files written out -> what CDC data is contained. always easier for people to digest when leverage on existing knowledge (actions here) instead of bringing up new definitions

Reply:
> always easier for people to digest when leverage on existing knowledge (actions here) instead of bringing up new definitions

Cann't agree more, let's not introduce new terminology.

Reply:
fixed. moved to a table in "Read" section.